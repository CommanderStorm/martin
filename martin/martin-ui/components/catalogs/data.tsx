"use client";

import {
	Database,
	Eye,
	Globe,
	ImageIcon,
	Layers,
	Palette,
	Search,
	Type,
} from "lucide-react";
import { ErrorState, InlineErrorState } from "@/components/error/error-state";
import { CatalogSkeleton } from "@/components/loading/catalog-skeleton";
import { LoadingSpinner } from "@/components/loading/loading-spinner";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import {
	Card,
	CardContent,
	CardDescription,
	CardHeader,
	CardTitle,
} from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { Tooltip, TooltipContent, TooltipTrigger } from "../ui/tooltip";

interface DataSource {
	id: string;
	name: string;
	type: string;
	description: string;
	layers: number;
	lastUpdated: string;
	size: string;
}

interface DataCatalogProps {
	dataSources: DataSource[];
	searchQuery: string;
	onSearchChangeAction: (query: string) => void;
	isLoading?: boolean;
	isSearching?: boolean;
	error?: Error | null;
	searchError?: Error | null;
	onRetry?: () => void;
	onRetrySearch?: () => void;
	isRetrying?: boolean;
}

export function DataCatalog({
	dataSources,
	searchQuery,
	onSearchChangeAction,
	isLoading = false,
	isSearching = false,
	error = null,
	searchError = null,
	onRetry,
	onRetrySearch,
	isRetrying = false,
}: DataCatalogProps) {
	if (isLoading) {
		return (
			<CatalogSkeleton
				title="Data Sources Catalog"
				description="Manage and explore all available tile sources, sprites, and fonts"
			/>
		);
	}

	if (error) {
		return (
			<ErrorState
				title="Failed to Load Data Catalog"
				description="Unable to fetch data sources from the server"
				error={error}
				onRetry={onRetry}
				isRetrying={isRetrying}
				variant="server"
				showDetails={true}
			/>
		);
	}

	const filteredDataSources = dataSources.filter(
		(source) =>
			source.name.toLowerCase().includes(searchQuery.toLowerCase()) ||
			source.type.toLowerCase().includes(searchQuery.toLowerCase()),
	);

	const getIcon = (type: string) => {
		switch (type) {
			case "Vector Tiles":
				return <Layers className="w-5 h-5 text-primary" />;
			case "Raster Tiles":
				return <ImageIcon className="w-5 h-5 text-primary" />;
			case "Sprites":
				return <Globe className="w-5 h-5 text-primary" />;
			case "Fonts":
				return <Type className="w-5 h-5 text-primary" />;
			default:
				return <Database className="w-5 h-5 text-primary" />;
		}
	};

	return (
		<div className="space-y-6">
			<div className="flex items-center justify-between">
				<div>
					<h2 className="text-2xl font-bold text-foreground">
						Data Sources Catalog
					</h2>
					<p className="text-muted-foreground">
						Manage and explore all available tile sources, sprites, and fonts
					</p>
				</div>
				<div className="relative">
					<Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 w-4 h-4" />
					<Input
						placeholder="Search data sources..."
						value={searchQuery}
						onChange={(e) => onSearchChangeAction(e.target.value)}
						className="pl-10 w-64 bg-card"
					/>
					{isSearching && (
						<div className="absolute right-3 top-1/2 transform -translate-y-1/2">
							<LoadingSpinner size="sm" />
						</div>
					)}
				</div>
			</div>

			{searchError && (
				<InlineErrorState
					message="Search failed. Please try again."
					onRetry={onRetrySearch}
					variant="network"
				/>
			)}

			{isSearching ? (
				<div className="flex items-center justify-center py-12">
					<div className="text-center">
						<LoadingSpinner size="lg" className="mx-auto mb-4" />
						<p className="text-muted-foreground">Searching data sources...</p>
					</div>
				</div>
			) : (
				<div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6">
					{filteredDataSources.map((source) => (
						<Card key={source.id} className="hover:shadow-lg transition-shadow">
							<CardHeader>
								<div className="flex items-center justify-between">
									<div className="flex items-center space-x-2">
										{getIcon(source.type)}
										<CardTitle className="text-lg">{source.name}</CardTitle>
									</div>
									<Badge variant="secondary">{source.type}</Badge>
								</div>
								<CardDescription>{source.description}</CardDescription>
							</CardHeader>
							<CardContent>
								<div className="space-y-2 text-sm text-muted-foreground">
									<div className="flex justify-between">
										<span>Layers:</span>
										<span>{source.layers}</span>
									</div>
									<div className="flex justify-between">
										<span>Size:</span>
										<span>{source.size}</span>
									</div>
									<div className="flex justify-between">
										<span>Updated:</span>
										<span>{source.lastUpdated}</span>
									</div>
								</div>
								<div className="flex space-x-2 mt-4">
  								<Tooltip>
                    <TooltipTrigger className="flex flex-1" asChild>
     									<Button
      										size="sm"
      										variant="outline"
      										className="flex-1 bg-transparent"
      										disabled
     									>
      										<Eye className="w-4 h-4 mr-2" />
      										Inspect
     									</Button>
                    </TooltipTrigger>
  									<TooltipContent>
                      <p>Not currently implemented in the frontend</p>
                    </TooltipContent>
                  </Tooltip>
  								<Tooltip>
                    <TooltipTrigger className="flex flex-1" asChild>
   									<Button
    										size="sm"
    										className="flex-1 bg-primary hover:bg-purple-700"
    										disabled
    								>
    										<Palette className="w-4 h-4 mr-2" />
    										Style
   									</Button>
    								</TooltipTrigger>
  									<TooltipContent>
                      <p>Not currently implemented in the frontend</p>
                    </TooltipContent>
                  </Tooltip>
								</div>
							</CardContent>
						</Card>
					))}
				</div>
			)}

			{!isSearching &&
				filteredDataSources.length === 0 &&
				searchQuery &&
				!searchError && (
					<div className="text-center py-12">
						<p className="text-muted-foreground">
							No data sources found matching "{searchQuery}"
						</p>
					</div>
				)}
		</div>
	);
}
