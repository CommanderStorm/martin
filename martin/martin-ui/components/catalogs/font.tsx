import { Download, Eye, Search, Type } from "lucide-react";
import { ErrorState } from "@/components/error/error-state";
import { CatalogSkeleton } from "@/components/loading/catalog-skeleton";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import type { Font } from "@/lib/types";
import { DisabledNonInteractiveButton } from "../ui/disabledNonInteractiveButton";
import { Tooltip, TooltipContent, TooltipTrigger } from "../ui/tooltip";

interface FontCatalogProps {
  fonts?: { [name: string]: Font };
  searchQuery?: string;
  onSearchChangeAction?: (query: string) => void;
  isLoading?: boolean;
  error?: Error | null;
  onRetry?: () => void;
  isRetrying?: boolean;
}

export function FontCatalog({
  fonts,
  searchQuery = "",
  onSearchChangeAction = () => {},
  isLoading,
  error = null,
  onRetry,
  isRetrying = false,
}: FontCatalogProps) {
  if (isLoading) {
    return <CatalogSkeleton description="Preview all available font assets" title="Font Catalog" />;
  }

  if (error) {
    return (
      <ErrorState
        description="Unable to fetch font catalog from the server"
        error={error}
        isRetrying={isRetrying}
        onRetry={onRetry}
        showDetails={true}
        title="Failed to Load Fonts"
        variant="server"
      />
    );
  }

  const filteredFonts = Object.entries(fonts || {}).filter(
    ([name, font]) =>
      name.toLowerCase().includes(searchQuery.toLowerCase()) ||
      font.family.toLowerCase().includes(searchQuery.toLowerCase()) ||
      font.style.toLowerCase().includes(searchQuery.toLowerCase()),
  );

  return (
    <div className="space-y-6">
      <div className="flex items-center justify-between">
        <div>
          <h2 className="text-2xl font-bold text-foreground">Font Catalog</h2>
          <p className="text-muted-foreground">Preview all available font assets</p>
        </div>
        <div className="relative">
          <Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 w-4 h-4" />
          <Input
            className="pl-10 w-64 bg-card"
            onChange={(e) => onSearchChangeAction(e.target.value)}
            placeholder="Search fonts..."
            value={searchQuery}
          />
        </div>
      </div>

      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6">
        {filteredFonts.map(([name, font]) => (
          <Card className="hover:shadow-lg transition-shadow" key={name}>
            <CardHeader>
              <div className="flex items-center justify-between">
                <div className="flex items-center space-x-2">
                  <Type className="w-5 h-5 text-primary" />
                  <CardTitle className="text-lg">{name}</CardTitle>
                </div>
                {font.format && (
                  <Badge className="uppercase" variant="secondary">
                    {font.format}
                  </Badge>
                )}
              </div>
              <CardDescription>
                Family: {font.family} â€¢ Style: {font.style}
              </CardDescription>
            </CardHeader>
            <CardContent>
              <div className="space-y-4">
                <div className="p-3 bg-gray-50 text-gray-900 rounded-lg">
                  <p className="text-sm font-medium mb-2 text-gray-900">Preview:</p>
                  <Tooltip>
                    <TooltipTrigger>
                      <p
                        className="text-base text-gray-900 blur-sm animate-pulse"
                        style={{ fontFamily: font.family, fontWeight: 500 }}
                      >
                        The quick brown fox jumps over the lazy dog
                      </p>
                    </TooltipTrigger>
                    <TooltipContent>
                      <p>Not currently implemented in the frontend</p>
                    </TooltipContent>
                  </Tooltip>
                </div>
                <div className="space-y-2 text-sm text-muted-foreground">
                  <div className="flex justify-between">
                    <span>Glyph count:</span>
                    <span>{font.glyphs}</span>
                  </div>
                </div>
                <div className="flex space-x-2">
                  <Button className="flex-1 bg-transparent" size="sm" variant="outline">
                    <Download className="w-4 h-4 mr-2" />
                    Download
                  </Button>
                  <Tooltip>
                    <TooltipTrigger className="flex-1 flex">
                      <DisabledNonInteractiveButton className="flex-1 grow" disabled size="sm">
                        <Eye className="w-4 h-4 mr-2" />
                        Details
                      </DisabledNonInteractiveButton>
                    </TooltipTrigger>
                    <TooltipContent>
                      <p>Not currently implemented in the frontend</p>
                    </TooltipContent>
                  </Tooltip>
                </div>
              </div>
            </CardContent>
          </Card>
        ))}
      </div>

      {filteredFonts.length === 0 && searchQuery && (
        <div className="text-center py-12">
          <p className="text-muted-foreground">No fonts found matching "{searchQuery}"</p>
        </div>
      )}
    </div>
  );
}
