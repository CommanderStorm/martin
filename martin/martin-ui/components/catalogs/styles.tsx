import { Brush, Download, Eye, Map, Search } from "lucide-react";
import { ErrorState } from "@/components/error/error-state";
import { CatalogSkeleton } from "@/components/loading/catalog-skeleton";
import { Badge } from "@/components/ui/badge";
import { Button } from "@/components/ui/button";
import {
	Card,
	CardContent,
	CardDescription,
	CardHeader,
	CardTitle,
} from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import { DisabledNonInteractiveButton } from "../ui/disabledNonInteractiveButton";
import { Tooltip, TooltipContent, TooltipTrigger } from "../ui/tooltip";
import type { Style } from "@/lib/types";

interface StylesCatalogProps {
	styles?: {[name: string]: Style};
	searchQuery?: string;
	onSearchChangeAction?: (query: string) => void;
	isLoading?: boolean;
	error?: Error | null;
	onRetry?: () => void;
	isRetrying?: boolean;
}

export function StylesCatalog({
	styles,
	searchQuery = "",
	onSearchChangeAction = () => {},
	isLoading,
	error = null,
	onRetry,
	isRetrying = false,
}: StylesCatalogProps) {
	if (isLoading) {
		return (
			<CatalogSkeleton
				title="Styles Catalog"
				description="Browse and preview all available map styles and themes"
			/>
		);
	}

	if (error) {
		return (
			<ErrorState
				title="Failed to Load Styles"
				description="Unable to fetch style catalog from the server"
				error={error}
				onRetry={onRetry}
				isRetrying={isRetrying}
				variant="server"
				showDetails={true}
			/>
		);
	}

	const filteredStyles = Object.entries(styles ||{}).filter(
		([name,style]) =>
			name.toLowerCase().includes(searchQuery.toLowerCase()) ||
			style.description.toLowerCase().includes(searchQuery.toLowerCase()) ||
			style.type.toLowerCase().includes(searchQuery.toLowerCase()),
	);

	return (
		<div className="space-y-6">
			<div className="flex items-center justify-between">
				<div>
					<h2 className="text-2xl font-bold text-foreground">Styles Catalog</h2>
					<p className="text-muted-foreground">
						Browse and preview all available map styles and themes
					</p>
				</div>
				<div className="relative">
					<Search className="absolute left-3 top-1/2 transform -translate-y-1/2 text-gray-400 w-4 h-4" />
					<Input
						placeholder="Search styles..."
						className="pl-10 w-64 bg-card"
						value={searchQuery}
						onChange={(e) => onSearchChangeAction(e.target.value)}
					/>
				</div>
			</div>

			<div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6">
				{filteredStyles.map(([name,style]) => (
					<Card key={name} className="hover:shadow-lg transition-shadow">
						<CardHeader>
							<div className="flex items-center justify-between">
								<div className="flex items-center space-x-2">
									<Brush className="w-5 h-5 text-primary" />
									<CardTitle className="text-lg">{name}</CardTitle>
								</div>
								<Badge variant="secondary">{style.type}</Badge>
							</div>
							<CardDescription>{style.description}</CardDescription>
						</CardHeader>
						<CardContent>
							<div className="space-y-4">
								<div className="p-3 aspect-video rounded-lg bg-gradient-to-br from-gray-200 to-gray-300 flex items-center justify-center relative overflow-hidden">
									<div className="absolute inset-0 opacity-20">
										<div
											className="w-full h-full bg-gradient-to-r"
											style={{
												background: `linear-gradient(45deg, ${style.colors.join(", ")})`,
											}}
										></div>
									</div>
									<Map className="w-8 h-8 text-gray-600 z-10" />
								</div>
								<div className="space-y-2 text-sm text-muted-foreground">
									<div className="flex justify-between">
										<span>Version:</span>
										<span>{style.version}</span>
									</div>
									<div className="flex justify-between">
										<span>Layers:</span>
										<span>{style.layers}</span>
									</div>
									<div className="flex justify-between">
										<span>Usage:</span>
										<span>{style.usage}</span>
									</div>
									<div className="flex justify-between">
										<span>Modified:</span>
										<span>{style.lastModified}</span>
									</div>
								</div>
								<div>
									<p className="text-sm font-medium mb-2">Color Palette:</p>
									<div className="flex space-x-1">
										{style.colors.map((color, i) => (
											<div
												key={i}
												className="w-6 h-6 rounded border border-gray-200"
												style={{ backgroundColor: color }}
												title={color}
											></div>
										))}
									</div>
								</div>
								<div className="flex space-x-2">
									<Button
										size="sm"
										variant="outline"
										className="flex-1 bg-transparent"
									>
										<Download className="w-4 h-4 mr-2" />
										Download
									</Button>

									<Tooltip>
										<TooltipTrigger className="flex flex-1">
											<DisabledNonInteractiveButton
												size="sm"
												className="flex-1"
											>
												<Eye className="w-4 h-4 mr-2" />
												Preview
											</DisabledNonInteractiveButton>
										</TooltipTrigger>
										<TooltipContent>
											<p>Not currently implemented in the frontend</p>
										</TooltipContent>
									</Tooltip>
								</div>
							</div>
						</CardContent>
					</Card>
				))}
			</div>

			{filteredStyles.length === 0 && searchQuery && (
				<div className="text-center py-12">
					<p className="text-muted-foreground">
						No styles found matching "{searchQuery}"
					</p>
				</div>
			)}
		</div>
	);
}
