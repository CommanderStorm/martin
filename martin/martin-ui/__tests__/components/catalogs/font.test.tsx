import { fireEvent, render, screen } from "@testing-library/react";
import type React from "react";
import { FontCatalog } from "@/components/catalogs/font";
import type { Font } from "@/lib/types";

// Mock all dependencies
jest.mock("@/components/error/error-state", () => ({
  ErrorState: ({ title, description }: { title: string; description: string }) => (
    <div data-testid="error-state">
      <div data-testid="error-title">{title}</div>
      <div data-testid="error-description">{description}</div>
    </div>
  ),
}));

jest.mock("@/components/loading/catalog-skeleton", () => ({
  CatalogSkeleton: ({ title, description }: { title: string; description: string }) => (
    <div data-testid="catalog-skeleton">
      <div data-testid="skeleton-title">{title}</div>
      <div data-testid="skeleton-description">{description}</div>
    </div>
  ),
}));

// Mock UI components to avoid tooltip provider issues
jest.mock("@/components/ui/tooltip", () => ({
  Tooltip: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
  TooltipContent: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="tooltip-content">{children}</div>
  ),
  TooltipTrigger: ({ children }: { children: React.ReactNode }) => <div>{children}</div>,
}));

jest.mock("@/components/ui/button", () => ({
  Button: ({ asChild, children, ...props }: any) => <button {...props}>{children}</button>,
}));

jest.mock("@/components/ui/copy-link-button", () => ({
  CopyLinkButton: ({ toastMessage, children, ...props }: any) => (
    <button data-testid="copy-link-button" {...props}>{children ?? "Copy Link"}</button>
  ),
}));

jest.mock("@/components/ui/badge", () => ({
  Badge: ({ children, ...props }: any) => (
    <span data-testid="badge" {...props}>
      {children}
    </span>
  ),
}));

jest.mock("@/components/ui/input", () => ({
  Input: (props: any) => <input {...props} />,
}));

jest.mock("@/components/ui/card", () => ({
  Card: ({ children, ...props }: any) => <div {...props}>{children}</div>,
  CardContent: ({ children, ...props }: any) => (
    <div data-testid="card-content" {...props}>
      {children}
    </div>
  ),
  CardDescription: ({ children, ...props }: any) => (
    <div data-testid="card-description" {...props}>
      {children}
    </div>
  ),
  CardHeader: ({ children, ...props }: any) => (
    <div data-testid="card-header" {...props}>
      {children}
    </div>
  ),
  CardTitle: ({ children, ...props }: any) => (
    <div data-testid="card-title" {...props}>
      {children}
    </div>
  ),
}));

jest.mock("@/components/ui/disabledNonInteractiveButton", () => ({
  DisabledNonInteractiveButton: ({ children, ...props }: any) => (
    <button {...props} disabled>
      {children}
    </button>
  ),
}));

jest.mock("lucide-react", () => ({
  Download: () => <div data-testid="download-icon">Download</div>,
  Eye: () => <div data-testid="eye-icon">Eye</div>,
  Search: () => <div data-testid="search-icon">Search</div>,
  Type: () => <div data-testid="type-icon">Type</div>,
}));

describe("FontCatalog Component", () => {
  const mockFonts: { [name: string]: Font } = {
    "Noto Sans Bold": {
      end: 255,
      family: "Noto Sans",
      format: "ttf",
      glyphs: 380,
      lastModifiedAt: new Date("2023-03-20"),
      start: 0,
      style: "Bold",
    },
    "Open Sans Regular": {
      end: 255,
      family: "Open Sans",
      format: "otf",
      glyphs: 420,
      lastModifiedAt: new Date("2023-02-15"),
      start: 0,
      style: "Regular",
    },
    "Roboto Medium": {
      end: 255,
      family: "Roboto",
      format: "ttf",
      glyphs: 350,
      lastModifiedAt: new Date("2023-01-01"),
      start: 0,
      style: "Medium",
    },
  };

  const defaultProps = {
    error: null,
    fonts: mockFonts,
    isLoading: false,
    isRetrying: false,
    onRetry: jest.fn(),
    onSearchChangeAction: jest.fn(),
    searchQuery: "",
  };

  it("matches snapshot for loading state", () => {
    const { asFragment } = render(<FontCatalog {...defaultProps} isLoading={true} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("matches snapshot for loaded state with mock data", () => {
    const { asFragment } = render(<FontCatalog {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders loading skeleton when isLoading is true", () => {
    render(<FontCatalog {...defaultProps} isLoading={true} />);
    expect(screen.getByTestId("catalog-skeleton")).toBeInTheDocument();
    expect(screen.getByTestId("skeleton-title").textContent).toBe("Font Catalog");
    expect(screen.getByTestId("skeleton-description").textContent).toBe(
      "Preview all available font assets",
    );
  });

  it("renders error state when there is an error", () => {
    const error = new Error("Test error");
    render(<FontCatalog {...defaultProps} error={error} />);
    expect(screen.getByTestId("error-state")).toBeInTheDocument();
    expect(screen.getByTestId("error-title").textContent).toBe("Failed to Load Fonts");
  });

  it("renders font catalog correctly", () => {
    render(<FontCatalog {...defaultProps} />);
    expect(screen.getByText("Font Catalog")).toBeInTheDocument();

    // Get all card headers
    const headers = screen.getAllByTestId("card-header");
    expect(headers.length).toBe(3);

    // Check that each font name is displayed
    expect(screen.getByText("Roboto Medium")).toBeInTheDocument();
    expect(screen.getByText("Open Sans Regular")).toBeInTheDocument();
    expect(screen.getByText("Noto Sans Bold")).toBeInTheDocument();

    // Verify format badges are displayed
    const badges = screen.getAllByTestId("badge");
    expect(badges.length).toBe(3);
    expect(badges[0].textContent).toBe("ttf");
    expect(badges[1].textContent).toBe("otf");
    expect(badges[2].textContent).toBe("ttf");

    // Verify glyph counts are displayed
    expect(screen.getByText("350")).toBeInTheDocument();
    expect(screen.getByText("420")).toBeInTheDocument();
    expect(screen.getByText("380")).toBeInTheDocument();

    // Verify family and style information is displayed
    expect(screen.getByText("Family: Roboto • Style: Medium")).toBeInTheDocument();
    expect(screen.getByText("Family: Open Sans • Style: Regular")).toBeInTheDocument();
    expect(screen.getByText("Family: Noto Sans • Style: Bold")).toBeInTheDocument();
  });

  it("filters fonts based on search query", () => {
    render(<FontCatalog {...defaultProps} searchQuery="roboto" />);

    // Should only show the Roboto font
    const headers = screen.getAllByTestId("card-header");
    expect(headers.length).toBe(1);
    expect(screen.getByText("Roboto Medium")).toBeInTheDocument();
    expect(screen.queryByText("Open Sans Regular")).not.toBeInTheDocument();
    expect(screen.queryByText("Noto Sans Bold")).not.toBeInTheDocument();
  });

  it("filters fonts based on font family", () => {
    render(<FontCatalog {...defaultProps} searchQuery="open" />);

    // Should only show the Open Sans font
    const headers = screen.getAllByTestId("card-header");
    expect(headers.length).toBe(1);
    expect(screen.queryByText("Roboto Medium")).not.toBeInTheDocument();
    expect(screen.getByText("Open Sans Regular")).toBeInTheDocument();
    expect(screen.queryByText("Noto Sans Bold")).not.toBeInTheDocument();
  });

  it("filters fonts based on style", () => {
    render(<FontCatalog {...defaultProps} searchQuery="bold" />);

    // Should only show the Noto Sans Bold font
    const headers = screen.getAllByTestId("card-header");
    expect(headers.length).toBe(1);
    expect(screen.queryByText("Roboto Medium")).not.toBeInTheDocument();
    expect(screen.queryByText("Open Sans Regular")).not.toBeInTheDocument();
    expect(screen.getByText("Noto Sans Bold")).toBeInTheDocument();
  });

  it("shows no results message when search has no matches", () => {
    render(<FontCatalog {...defaultProps} searchQuery="nonexistent" />);
    expect(screen.getByText(/No fonts found matching "nonexistent"/i)).toBeInTheDocument();

    // Should not render any cards
    const headers = screen.queryAllByTestId("card-header");
    expect(headers.length).toBe(0);
  });

  it("calls onSearchChangeAction when search input changes", () => {
    render(<FontCatalog {...defaultProps} />);
    const searchInput = screen.getByPlaceholderText("Search fonts...");

    fireEvent.change(searchInput, { target: { value: "new search" } });
    expect(defaultProps.onSearchChangeAction).toHaveBeenCalledWith("new search");
  });

  it("renders download and details buttons for each font", () => {
    render(<FontCatalog {...defaultProps} />);

    // We should have 3 copy link buttons (one for each font)
    const copyLinkButtons = screen.getAllByTestId("copy-link-button");
    expect(copyLinkButtons.length).toBe(3);

    // We should have 3 eye icons for details (one for each font)
    const eyeIcons = screen.getAllByTestId("eye-icon");
    expect(eyeIcons.length).toBe(3);
  });
});
