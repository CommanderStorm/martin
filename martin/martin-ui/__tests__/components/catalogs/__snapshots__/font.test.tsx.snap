// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`FontCatalog Component matches snapshot for loaded state with mock data 1`] = `
<DocumentFragment>
  <div
    class="space-y-6"
  >
    <div
      class="flex items-center justify-between"
    >
      <div>
        <h2
          class="text-2xl font-bold text-foreground"
        >
          Font Catalog
        </h2>
        <p
          class="text-muted-foreground"
        >
          Preview all available font assets
        </p>
      </div>
      <div
        class="relative"
      >
        <div
          data-testid="search-icon"
        >
          Search
        </div>
        <input
          class="pl-10 w-64 bg-card"
          placeholder="Search fonts..."
          value=""
        />
      </div>
    </div>
    <div
      class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-6"
    >
      <div
        class="hover:shadow-lg transition-shadow"
      >
        <div
          data-testid="card-header"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                data-testid="type-icon"
              >
                Type
              </div>
              <div
                class="text-lg"
                data-testid="card-title"
              >
                Noto Sans Bold
              </div>
            </div>
            <span
              class="uppercase"
              data-testid="badge"
              variant="secondary"
            >
              ttf
            </span>
          </div>
          <div
            data-testid="card-description"
          >
            Family: Noto Sans • Style: Bold
          </div>
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="space-y-4"
          >
            <div
              class="p-3 bg-gray-50 text-gray-900 rounded-lg"
            >
              <p
                class="text-sm font-medium mb-2 text-gray-900"
              >
                Preview:
              </p>
              <div>
                <div>
                  <p
                    class="text-base text-gray-900 blur-sm animate-pulse"
                    style="font-family: Noto Sans; font-weight: 500;"
                  >
                    The quick brown fox jumps over the lazy dog
                  </p>
                </div>
                <div
                  data-testid="tooltip-content"
                >
                  <p>
                    Not currently implemented in the frontend
                  </p>
                </div>
              </div>
            </div>
            <div
              class="space-y-2 text-sm text-muted-foreground"
            >
              <div
                class="flex justify-between"
              >
                <span>
                  Glyph count:
                </span>
                <span>
                  380
                </span>
              </div>
            </div>
            <div
              class="flex space-x-2"
            >
              <button
                class="flex-1 bg-transparent"
                data-testid="copy-link-button"
                link="/font/Noto Sans Bold/{range}"
                variant="outline"
              >
                Copy Link
              </button>
              <div>
                <div>
                  <button
                    class="flex-1 grow"
                    disabled=""
                  >
                    <div
                      data-testid="eye-icon"
                    >
                      Eye
                    </div>
                    Details
                  </button>
                </div>
                <div
                  data-testid="tooltip-content"
                >
                  <p>
                    Not currently implemented in the frontend
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="hover:shadow-lg transition-shadow"
      >
        <div
          data-testid="card-header"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                data-testid="type-icon"
              >
                Type
              </div>
              <div
                class="text-lg"
                data-testid="card-title"
              >
                Open Sans Regular
              </div>
            </div>
            <span
              class="uppercase"
              data-testid="badge"
              variant="secondary"
            >
              otf
            </span>
          </div>
          <div
            data-testid="card-description"
          >
            Family: Open Sans • Style: Regular
          </div>
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="space-y-4"
          >
            <div
              class="p-3 bg-gray-50 text-gray-900 rounded-lg"
            >
              <p
                class="text-sm font-medium mb-2 text-gray-900"
              >
                Preview:
              </p>
              <div>
                <div>
                  <p
                    class="text-base text-gray-900 blur-sm animate-pulse"
                    style="font-family: Open Sans; font-weight: 500;"
                  >
                    The quick brown fox jumps over the lazy dog
                  </p>
                </div>
                <div
                  data-testid="tooltip-content"
                >
                  <p>
                    Not currently implemented in the frontend
                  </p>
                </div>
              </div>
            </div>
            <div
              class="space-y-2 text-sm text-muted-foreground"
            >
              <div
                class="flex justify-between"
              >
                <span>
                  Glyph count:
                </span>
                <span>
                  420
                </span>
              </div>
            </div>
            <div
              class="flex space-x-2"
            >
              <button
                class="flex-1 bg-transparent"
                data-testid="copy-link-button"
                link="/font/Open Sans Regular/{range}"
                variant="outline"
              >
                Copy Link
              </button>
              <div>
                <div>
                  <button
                    class="flex-1 grow"
                    disabled=""
                  >
                    <div
                      data-testid="eye-icon"
                    >
                      Eye
                    </div>
                    Details
                  </button>
                </div>
                <div
                  data-testid="tooltip-content"
                >
                  <p>
                    Not currently implemented in the frontend
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="hover:shadow-lg transition-shadow"
      >
        <div
          data-testid="card-header"
        >
          <div
            class="flex items-center justify-between"
          >
            <div
              class="flex items-center space-x-2"
            >
              <div
                data-testid="type-icon"
              >
                Type
              </div>
              <div
                class="text-lg"
                data-testid="card-title"
              >
                Roboto Medium
              </div>
            </div>
            <span
              class="uppercase"
              data-testid="badge"
              variant="secondary"
            >
              ttf
            </span>
          </div>
          <div
            data-testid="card-description"
          >
            Family: Roboto • Style: Medium
          </div>
        </div>
        <div
          data-testid="card-content"
        >
          <div
            class="space-y-4"
          >
            <div
              class="p-3 bg-gray-50 text-gray-900 rounded-lg"
            >
              <p
                class="text-sm font-medium mb-2 text-gray-900"
              >
                Preview:
              </p>
              <div>
                <div>
                  <p
                    class="text-base text-gray-900 blur-sm animate-pulse"
                    style="font-family: Roboto; font-weight: 500;"
                  >
                    The quick brown fox jumps over the lazy dog
                  </p>
                </div>
                <div
                  data-testid="tooltip-content"
                >
                  <p>
                    Not currently implemented in the frontend
                  </p>
                </div>
              </div>
            </div>
            <div
              class="space-y-2 text-sm text-muted-foreground"
            >
              <div
                class="flex justify-between"
              >
                <span>
                  Glyph count:
                </span>
                <span>
                  350
                </span>
              </div>
            </div>
            <div
              class="flex space-x-2"
            >
              <button
                class="flex-1 bg-transparent"
                data-testid="copy-link-button"
                link="/font/Roboto Medium/{range}"
                variant="outline"
              >
                Copy Link
              </button>
              <div>
                <div>
                  <button
                    class="flex-1 grow"
                    disabled=""
                  >
                    <div
                      data-testid="eye-icon"
                    >
                      Eye
                    </div>
                    Details
                  </button>
                </div>
                <div
                  data-testid="tooltip-content"
                >
                  <p>
                    Not currently implemented in the frontend
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FontCatalog Component matches snapshot for loading state 1`] = `
<DocumentFragment>
  <div
    data-testid="catalog-skeleton"
  >
    <div
      data-testid="skeleton-title"
    >
      Font Catalog
    </div>
    <div
      data-testid="skeleton-description"
    >
      Preview all available font assets
    </div>
  </div>
</DocumentFragment>
`;
